### PR TITLE
Unquote text and identifiers in PPL parsing

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Following #33 , StringUtils is copied from sql repo to ppl-spark-integration

### Issues Resolved
* #260 
* #130 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanBasicQueriesTranslatorTestSuite.scala [hunk 1]
```diff
@@ -32,7 +32,16 @@ class PPLLogicalPlanBasicQueriesTranslatorTestSuite
     val projectList: Seq[NamedExpression] = Seq(UnresolvedStar(None))
     val expectedPlan = Project(projectList, UnresolvedRelation(Seq("table")))
     assertEquals(expectedPlan, logPlan)
+  }
+
+  test("test simple search with escaped table name") {
+    // if successful build ppl logical plan and translate to catalyst logical plan
+    val context = new CatalystPlanContext
+    val logPlan = planTrnasformer.visit(plan(pplParser, "source=`table`", false), context)
```
Comment (dai-chen) [inline ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanBasicQueriesTranslatorTestSuite.scala:40]: np: typo in `planTrnasformer`

File: ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ */
+
+package org.opensearch.sql.common.utils;
+
+import com.google.common.base.Strings;
+
+import java.util.IllegalFormatException;
+import java.util.Locale;
+
+public class StringUtils {
```
Comment (dai-chen) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java:13]: Any UT in SQL repo we can copy for this?

File: integ-test/src/test/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala [hunk 3]
```diff
@@ -65,6 +65,33 @@ class FlintSparkPPLBasicITSuite
...
+    // Retrieve the results
+    val results: Array[Row] = frame.collect()
+    // Define the expected results
+    val expectedResults: Array[Row] = Array(
+      Row("Jake", 70, "California", "USA", 2023, 4),
+      Row("Hello", 30, "New York", "USA", 2023, 4),
+      Row("John", 25, "Ontario", "Canada", 2023, 4),
+      Row("Jane", 20, "Quebec", "Canada", 2023, 4))
+    // Compare the results
+    // Compare the results
```
Comment (ykmr1224) [inline integ-test/src/test/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:80]: Duplicate comment.
Looks like most of the comments are not needed, btw.

File: ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanBasicQueriesTranslatorTestSuite.scala (hunk 1 above)
Comment (seankao-az) [reply ppl-spark-integration/src/test/scala/org/opensearch/flint/spark/ppl/PPLLogicalPlanBasicQueriesTranslatorTestSuite.scala:40]: hmm this typo is everywhere. will fix all occurrence of it in this PR as well

File: ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (seankao-az) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java:13]: same comment. see another thread

Comment (seankao-az) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java:13]: There's a StringUtilsTest in SQL repo. Do we want to copy that to this repo as well?

File: integ-test/src/test/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala (hunk 3 above)
Comment (seankao-az) [reply integ-test/src/test/scala/org/opensearch/flint/spark/ppl/FlintSparkPPLBasicITSuite.scala:80]: same comments are prevalent in all PPL suites. Will keep the comments but delete the duplicate ones

File: ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java (hunk 2 above)
Comment (ykmr1224) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/common/utils/StringUtils.java:13]: Considering we might want to change the code, we should copy the unit test as well.

**Review by penghuo — APPROVED** [review]
Thx!

File: build.sbt [hunk 4]
```diff
@@ -127,6 +127,8 @@ lazy val pplSparkIntegration = (project in file("ppl-spark-integration"))
       "org.scalatest" %% "scalatest" % "3.2.15" % "test",
       "org.scalatest" %% "scalatest-flatspec" % "3.2.15" % "test",
       "org.scalatestplus" %% "mockito-4-6" % "3.2.15.0" % "test",
+      "org.junit.jupiter" % "junit-jupiter-api" % "5.9.0" % "test",
+      "org.junit.jupiter" % "junit-jupiter-engine" % "5.9.0" % "test",
```
Comment (ykmr1224) [inline build.sbt:131]: Those are for JUnit5. I think we don't want to mix the usage of JUnit4 and JUnit5.

Comment (seankao-az) [reply build.sbt:131]: I see that mixing JUnit 4 and 5 seems prevalent in the repo already..
but sure, not for ppl project yet.
will remove JUnit 5.

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Review by ykmr1224 — APPROVED** [review]

**Pull request merged**